### PR TITLE
Enhance code quality

### PR DESCRIPTION
**Pull request by RaiseYI**
Refactor and add unit tests for critical functions in `apps/main/src/index.ts` and `apps/main/src/init.ts`.

* **Unit Tests:**
  - Add `apps/main/src/index.test.ts` to test critical functions in `index.ts`.
  - Add `apps/main/src/init.test.ts` to test critical functions in `init.ts`.

* **Code Formatting:**
  - Modify `.prettierrc.mjs` to update code formatting rules.
  - Modify `.vscode/extensions.json` to add recommended extensions for consistent code formatting and linting.
  - Modify `.vscode/settings.json` to ensure all files adhere to the rules specified in `.editorconfig`.

---

For more details, open the [Copilot Workspace session](https://copilot-workspace.githubnext.com/RSSNext/Follow?shareId=f30d0dbf-4f1d-44d0-b2bf-350fcf847b28).

### CODE REVIEW

File: .prettierrc.mjs [hunk 1]
```diff
@@ -1,7 +1,9 @@
...
-  singleQuote: false,
-  printWidth: 100,
+  singleQuote: true,
+  printWidth: 80,
   tabWidth: 2,
-  trailingComma: "all",
+  trailingComma: 'es5',
+  bracketSpacing: true,
+  arrowParens: 'avoid',
 }
```
Comment (hyoban) [inline .prettierrc.mjs:9]: No, I don’t think we need to discuss and change code formatting options now.

File: .vscode/extensions.json [hunk 2]
```diff
@@ -1,3 +1,7 @@
 {
-  "recommendations": ["dbaeumer.vscode-eslint"]
+  "recommendations": [
+    "dbaeumer.vscode-eslint",
+    "esbenp.prettier-vscode",
+    "editorconfig.editorconfig"
```
Comment (Innei) [inline .vscode/extensions.json:5]: We don't need this

File: .prettierrc.mjs [hunk 3]
```diff
@@ -1,7 +1,9 @@
 export default {
   semi: false,
-  singleQuote: false,
-  printWidth: 100,
+  singleQuote: true,
```
Comment (Innei) [inline .prettierrc.mjs:3]: We have specified a unified code format, you do not need to modify it according to your own preferences.

File: apps/main/src/index.test.ts [hunk 4]
```diff
@@ -0,0 +1,114 @@
+import { app, BrowserWindow } from "electron";
+import { initializeAppStage0, initializeAppStage1 } from "./init";
+import { createMainWindow, getMainWindow } from "./window";
+import { handleUrlRouting } from "./lib/router";
+import { setAuthSessionToken, updateNotificationsToken } from "./lib/user";
+import { callWindowExpose } from "@follow/shared/bridge";
+import { env } from "@follow/shared/env";
+import { URL } from "url";
+
+jest.mock("electron", () => ({
```
Comment (Innei) [inline apps/main/src/index.test.ts:10]: Why jest?

**Comment by Innei** [comment]
This PR lacks rationality, it goes beyond the scope of this project, so it has been closed. If there are any issues, it can be reopened.

**Pull request closed without merging**